test/NuGet.Tests.Apex/NuGet.Tests.Apex/NuGetEndToEndTests/NuGetConsoleTestCase.cs

```diff
@@ -911,6 +911,42 @@ public async Task InstallLatestPackageInPMC(ProjectTemplate projectTemplate)
             }
         }
 
+        [NuGetWpfTheory]
+        [MemberData(nameof(GetPackagesConfigTemplates))]
+        public void VerifyInitScriptsExecution(ProjectTemplate projectTemplate)
+        {
+            EnsureVisualStudioHost();
+            using (var testContext = new ApexTestContext(VisualStudio, projectTemplate, XunitLogger))
+            {
+                // Arrange
+                SolutionService solutionService = VisualStudio.Get<SolutionService>();
+                var nugetConsole = GetConsole(testContext.Project);
+                var projectPath = testContext.Project.FullPath;
+                var source = "https://api.nuget.org/v3/index.json";
+
+                // Act
+                nugetConsole.Execute($"install-package EntityFramework -source {source} -Verbose");
+
+                // Assert
+                Assert.True(nugetConsole.IsMessageFoundInPMC("init.ps1"), "The init.ps1 script in TestProject was not executed when the EntityFramework package was installed");
+
+                // Act
+                solutionService.CreateEmptySolution();
+                ProjectTestExtension project2 = solutionService.AddProject(ProjectLanguage.CSharp, ProjectTemplate.ClassLibrary, ProjectTargetFramework.V48, "TestProject2");
+                nugetConsole.Execute($"install-package jquery -source {source} -Verbose");
+
+                // Assert
+                Assert.True(nugetConsole.IsMessageFoundInPMC("install.ps1"), "The install.ps1 script in TestProject2 was not executed when the jquery package was installed.");
+
+                // Act
+                solutionService.OpenProject(projectPath);
+                nugetConsole.Execute($"install-package entityframework.sqlservercompact -source {source} -Verbose");
+
+                // Assert
+                Assert.True(nugetConsole.IsMessageFoundInPMC("Install.ps1"), "The Install.ps1 script in TestProject was not executed when the Entityframework.sqlservercompact package was installed.");
```

Comment:
Is `nugetConsole.IsMessageFoundInPMC` case sensitive?  Because if not, then installing entityframework.sqlservercompact is not necessary to make the test pass, as the Assert above on line 939, following the install of of jquery, will satisfy the assert. If `IsMessageFoundInPMC` is case sensitive, there should be a comment, because it's really not obvious, and this looks like a test bug.

I also don't understand the purpose of testing two different packages with install.ps1 and checking the PMC output contains "install.ps1" both times.  What kind of bug do you hope to catch that a single install.ps1 won't catch?

Reply:
> > Is `nugetConsole.IsMessageFoundInPMC` case sensitive? Because if not, then installing entityframework.sqlservercompact is not necessary to make the test pass, as the Assert above on line 939, following the install of of jquery, will satisfy the assert. If `IsMessageFoundInPMC` is case sensitive, there should be a comment, because it's really not obvious, and this looks like a test bug.
> > I also don't understand the purpose of testing two different packages with install.ps1 and checking the PMC output contains "install.ps1" both times. What kind of bug do you hope to catch that a single install.ps1 won't catch?
> 
> Hi @zivkan, `nugetConsole.IsMessageFoundInPMC` is case sensitive, I will add a comment to the case.
> 
> This case is what Ashish Jain asked us to add to the manual test earlier (as below), this case verify the scripts had been executed successfully after install a package (as below step 10). When installing the packages (jquery & entityframework.sqlservercompact), the PMC only contains install.ps1, so I add the code in 939 & 946 lines. ![image](https://user-images.githubusercontent.com/84695155/239840559-447c2a4b-e6aa-42e2-9f5d-1a83de333242.png)

Hi @zivkan, I have added a comment to `nugetConsole.IsMessageFoundInPMC` and this case have passed in corresponding build, could you please help to review this PR again, if you have any suggestion or concern please feel free let me know, thanks!